app/controllers/users_controller.rb

```diff
@@ -32,6 +32,9 @@ def update
     if user.save
       flash[:notice] = "Your information has been updated."
       render :show
+    elsif user.errors.full_messages == ["Email has already been taken"]
```

Comment:
DF, if you could give me a BRIEF rundown of this line - i know what's happening, but I'm curious how the chain "user.errors.full_messages" works. -- more specifically how are we calling '.errors'? 

..you're making it look too easy!